dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitorTest.kt

```diff
@@ -82,7 +85,50 @@ internal class DatadogRumMonitorTest {
     }
 
     @Test
-    fun `stopView doesn't send anything without matching startView`(
+    fun `startView sends previous unstopped view Rum Event`(
+        forge: Forge
+    ) {
+        val timestamp = forge.aTimestamp()
+        val key = forge.anAsciiString()
+        val name = forge.aStringMatching("[a-z]+(\\.[a-z]+)+")
+        val attributes = forge.exhaustiveAttributes()
+        var viewId: UUID? = null
+        whenever(mockTimeProvider.getServerTimestamp()) doReturn timestamp
+
+        val duration = measureNanoTime {
+            testedMonitor.startView(key, name, attributes)
+            viewId = GlobalRum.getRumContext().viewId
+            testedMonitor.startView(
+                forge.anAsciiString(),
+                forge.aStringMatching("[a-z]+(\\.[a-z]+)+"),
+                emptyMap()
+            )
+        }
+
+        checkNotNull(viewId)
+        argumentCaptor<RumEvent> {
+            verify(mockWriter).write(capture())
+
+            assertThat(lastValue)
```

Comment:
hmmm ...am I missing something but shouldn't be the **firstValue** here ? First we stop the first view and then we dispatch the second ?

Reply:
the mockWriter is called only once, so the firstValue and lastValue are the same here.